### PR TITLE
Add minCpuPlatform parameter.

### PR DESCRIPTION
**Pull request by johnlabarge**

### CODE REVIEW

File: src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java [hunk 1]
```diff
@@ -561,6 +565,30 @@ public FormValidation doCheckMachineType(@QueryParameter String value) {
...
 
+        public ListBoxModel doFillMinCpuPlatformItems(@AncestorInPath Jenkins context,
+                                               @QueryParameter("projectId") @RelativePath("..") final String projectId,
+                                               @QueryParameter("zone") final String zone,
+                                               @QueryParameter("credentialsId") @RelativePath("..") final String credentialsId)
+        {
+            ListBoxModel items = new ListBoxModel();
+            items.add("");
+            try {
+                ComputeClient compute = computeClient(context,credentialsId);
```
Comment (rachely3n) [inline src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java]: [nit] Space after the comma.

File: src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java [hunk 2]
```diff
@@ -561,6 +565,30 @@ public FormValidation doCheckMachineType(@QueryParameter String value) {
...
+                List<String> cpuPlatforms = compute.cpuPlatforms(projectId,zone);
```
Comment (rachely3n) [inline src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java]: Same as above.

File: src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java [hunk 3]
```diff
@@ -172,6 +172,19 @@ public int compare(MachineType o1, MachineType o2) {
         return machineTypes;
     }
 
+    public List<String> cpuPlatforms(String projectId, String zone) throws IOException {
+        List<String> cpuPlatforms = new ArrayList<String>();
+        String filter = String.format("selfLInk='%s'",zone);
```
Comment (rachely3n) [inline src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java]: Is filter used? Can we safely remove filter?

File: src/main/resources/com/google/jenkins/plugins/computeengine/InstanceConfiguration/config.jelly [hunk 4]
```diff
@@ -44,6 +44,9 @@
             <f:entry field="preemptible" title="${%Preemptible?}">
                 <f:checkbox/>
             </f:entry>
+             <f:entry field="minCpuPlatform" title="${%Minimum Cpu Platform}">
```
Comment (rachely3n) [inline src/main/resources/com/google/jenkins/plugins/computeengine/InstanceConfiguration/config.jelly]: Can you add a help html file for minCpuPlatform? e.g. help-description.html, help-namePrefix.html

Comment (johnlabarge) [reply src/main/resources/com/google/jenkins/plugins/computeengine/InstanceConfiguration/config.jelly]: Hmm thought I did maybe it didn't get added.

Comment (johnlabarge) [reply src/main/resources/com/google/jenkins/plugins/computeengine/InstanceConfiguration/config.jelly]: dang

**Comment by johnlabarge** [comment]
Will squash after LGTM.

**Comment by viglesiasce** [comment]
Thanks @johnlabarge!!!

**Comment by rachely3n** [comment]
I've approved and merged. Closing now!

**Pull request closed without merging**